src/modules/setup.ts

```diff
@@ -0,0 +1,277 @@
+    } catch (err) {
+      const statusMessage = err.response.status ?
+        `: Status ${err.response.status}` : ''
+      log.error(`Error fetching typings info ${statusMessage} (try: ${tryCount})`)
+      throw err
+    }
+  }
+  try {
+    return retry(getTypingsInfo, retryOpts)
+  } catch (e) {
+    log.error('Unable to get typings info from vtex.builder-hub.')
+    return {}
+  }
+
+}
+
+const appTypingsURL = async (account: string, workspace: string, appName: string, appVersion: string, builder: string): Promise<string> => {
+  const appId = await resolveAppId(appName, appVersion)
+  const vendor = getVendor(appId)
+  if (isLinked({'version': appId})) {
+  return `https://${workspace}--${account}.${publicEndpoint()}/_v/private/typings/linked/v1/${appId}/${typingsPath}/${builder}`
+  }
+  return `http://${vendor}.vteximg.com.br/_v/public/typings/v1/${appId}/${typingsPath}/${builder}`
+}
+
+const appsWithTypingsURLs = async (builder: string, account: string, workspace: string, appDependencies: Record<string, any>) => {
+  const result: Record<string, any> = {}
+  for (const [appName, appVersion] of Object.entries(appDependencies)) {
+    try {
+      result[appName] = await appTypingsURL(account, workspace, appName, appVersion, builder)
+    } catch (e) {
+      log.error(`Unable to generate typings URL for ${appName}@${appVersion}.`)
+    }
+  }
+  return result
+}
+
+const runYarn = (relativePath: string) => {
+  log.info(`Running yarn in ${relativePath}`)
+  execSync(
+    `${yarnPath} --force --non-interactive`,
+    {stdio: 'inherit', cwd: resolvePath(root, `${relativePath}`)}
+  )
+  log.info('Finished running yarn')
+}
+
+const yarnAddESLint = (relativePath: string) => {
+  log.info(`Adding lint configs in ${relativePath}`)
+  const lintDeps = join(' ', values(mapObjIndexed((version, name) => `${name}@${version}`, addToPackageJson)))
+  execSync( `${yarnPath} add ${lintDeps} --dev`,
+    {stdio: 'inherit', cwd: resolvePath(root, `${relativePath}`)}
+  )
+}
+
+const getTypings = async (manifest: Manifest, account: string, workspace: string) => {
+  const typingsData = await typingsInfo(account, workspace)
+
+  const buildersWithInjectedDeps =
+    pipe(
+      prop('builders'),
+      mapObjIndexed(
+        (version: string, builder: string) => {
+          const builderTypingsData = prop(builder, typingsData)
+          if (builderTypingsData && has(version, builderTypingsData)) {
+            return ramdaPath([version, 'injectedDependencies'], builderTypingsData)
+          }
+          return null
+        }
+      ),
+      ramdaReject(isNil)
+    )(manifest)
+
+  const buildersWithAppDeps =
+    pipe(
+      mapObjIndexed(
+        (injectedDependencies) => merge(manifest.dependencies, injectedDependencies)
+      ),
+      ramdaReject(isEmpty)
+    )(buildersWithInjectedDeps as Record<string, any>)
+
+  await pipe(
+    mapObjIndexed(
+      async (appDeps: Record<string, any>, builder: string) => {
```

Comment:
this could be a named function outside getTypings, would be easier to read.

Reply:
thx!